src/components/pages/devices/flyouts/deviceDelete/deviceDelete.js

```diff
@@ -0,0 +1,142 @@
+// Copyright (c) Microsoft. All rights reserved.
+
+import React, { Component } from 'react';
+import { Observable } from 'rxjs';
+
+import { IoTHubManagerService } from 'services';
+import { svgs } from 'utilities';
+import {
+  Btn,
+  BtnToolbar,
+  ErrorMsg,
+  Flyout,
+  FlyoutHeader,
+  FlyoutTitle,
+  FlyoutCloseBtn,
+  FlyoutContent,
+  Indicator,
+  Svg
+} from 'components/shared';
+
+import './deviceDelete.css';
+
+export class DeviceDelete extends Component {
+  constructor(props) {
+    super(props);
+    this.state = {
+      physicalDevices: [],
+      containsSimulatedDevices: false,
+      isPending: false,
+      error: undefined,
+      successCount: 0,
+      changesApplied: false
+    };
+  }
+
+  componentDidMount() {
+    if (this.props.devices) {
+      this.populateDevicesState(this.props.devices);
+    }
+  }
+
+  componentWillReceiveProps(nextProps) {
+    if (nextProps.devices && (this.props.devices || []).length !== nextProps.devices.length) {
+      this.populateDevicesState(nextProps.devices);
+    }
+  }
+
+  componentWillUnmount() {
+    if (this.subscription) this.subscription.unsubscribe();
+  }
+
+  populateDevicesState = (devices) => {
+    const deviceArr = devices || [];
```

Comment:
You don't need this since you can set the default value in the method signature.
`populateDevicesState = (devices = []) => {`